### PR TITLE
Allow path objects for data-files

### PR DESCRIPTION
**Pull request by jaraco**
Ported from pypa/setuptools#4495 by @InvincibleRMC.

### CODE REVIEW

File: distutils/command/install_data.py [hunk 1]
```diff
@@ -56,6 +57,16 @@ def run(self):
                     )
                 (out, _) = self.copy_file(f, self.install_dir)
                 self.outfiles.append(out)
+            elif isinstance(f, Path):
```
Comment (jaraco) [inline distutils/command/install_data.py:60]: This covers the case that one of the data files is a simple pathlib.Path, but it doesn't cover the case if a `pathlib.Path` is used in a tuple (in the else block below).

Comment (InvincibleRMC) [reply distutils/command/install_data.py:60]: Ah you are right. Should I fix it or do you want to fix it?

Comment (jaraco) [reply distutils/command/install_data.py:60]: I'm on it. But thanks for offering.

Comment (jaraco) [reply distutils/command/install_data.py:60]: My thinking is that `convert_path` already wraps all the places where a Path might be encountered, so probably that should be expanded to account for `pathlib.Path`.

**Comment by jaraco** [comment]
Looks like I jumped the gun on the merge. I've backed out the merge and going to re-submit.

**Pull request merged**